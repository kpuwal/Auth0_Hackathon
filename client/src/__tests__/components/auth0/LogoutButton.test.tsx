import LogoutButton from '../../../components/auth/LogoutButton';
import { shallow, mount } from 'enzyme';
import sinon from 'sinon';

describe('Logout Button', () => {
  it('renders logout button', () => {
    const component = shallow(<LogoutButton />);
    expect(component).toMatchSnapshot();
  });

  it('simulates click events', () => {
    const onButtonClick = sinon.spy();
    const wrapper = mount((
      <button onClick={onButtonClick} />
    ));
    wrapper.find('button').simulate('click');
    expect(onButtonClick).toHaveProperty('callCount', 1);
  });
})
