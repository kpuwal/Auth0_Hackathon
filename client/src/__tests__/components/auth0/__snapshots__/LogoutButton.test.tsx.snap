// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Logout Button renders logout button 1`] = `
<button
  className="nav-button"
  id="activated"
  onClick={[Function]}
>
  Log Out
</button>
`;
